feat: развитие сервиса (спринт 11)

OxanaKutukova opened the pull request:
1. Изменение хранилища на DB
2. Получение жанра и рейтинга по идентификатору
3. Получение списка всех жанров и по идентификатору

OxanaKutukova committed:
feat: развитие сервиса (спринт 11)

OxanaKutukova committed:
fix: добавлена схема БД в README.md

Troitsa commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -1,45 +1,47 @@
     public User getUserById(int userId) {
-        final User user = userStorage.getUserById(userId);
-        if (user == null) {
+        final User resUser = userStorage.getUserById(userId);
```
много дублей этих пары строк кода, может вынести в метод типа checkUser, будем получать либо юзера, вылетать по экспешену

OxanaKutukova committed:
fix: 1) Вынесено добавление жанров к фильму в FilmDbStorage
2) Вынесен повторяющийся кода в FilmService в метод private Film getFilmWithCheckNull (int filmId)
3) Вынесен повторяющийся код в UserService в метод     private User getUserWithCheckNull (int userId)
